Remove __future__ imports

peterfpeterson opened the pull request:
`__future__` were only used in python 2 and no-op in current python 3 and the compatibility code can be removed. 

Please note that
```python
from __future__ import annotations
```
needs to stay until python decided the default behavior will change ([reference](https://docs.python.org/3/library/__future__.html)).

This was discovered when running
```sh
$ pip-audit --local
Found 1 known vulnerability in 1 package
Name   Version ID                  Fix Versions
------ ------- ------------------- ------------
future 1.0.0   GHSA-xqrq-4mgf-ff32

$ pixi tree -i future

future 1.0.0
└── seekpath 2.1.0
    └── euphonic 1.4.4
        └── mantid 6.13.1.1
            └── mr_reduction 2.11.0
```
While confirming that mantid wasn't using the `future` dependency, the references in this PR were found. `seekpath` needs be dealt with separately so we can remove `future` and address [GHSA-xqrq-4mgf-ff32](https://osv.dev/vulnerability/GHSA-xqrq-4mgf-ff32).



There is no associated issue, but this is related to [EWM12760](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=12760)

### To test:

This is a removal of unused functionality. The current tests should be sufficient.

<!--  GATEKEEPER: When squashing, remove the section from HERE...  -->
---

### Reviewer

**Your comments will be used as part of the gatekeeper process.** Comment clearly on what you have checked and tested during your review. Provide an audit trail for any changes requested.

As per the [review guidelines](http://developer.mantidproject.org/ReviewingAPullRequest.html):

- Is the code of an acceptable quality? ([Code standards](http://developer.mantidproject.org/Standards/)/[GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html))
- Has a thorough functional test been performed? Do the changes handle unexpected input/situations?
- Are appropriately scoped unit and/or system tests provided?
- Do the release notes conform to the [guidelines](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html) and describe the changes appropriately?
- Has the relevant (user and developer) documentation been added/updated?

### Gatekeeper

As per the [gatekeeping guidelines](https://developer.mantidproject.org/Gatekeeping.html):

- Has a thorough first line review been conducted, including functional testing?
- At a high-level, is the code quality sufficient?
- Are the base, milestone and labels correct?
<!--  GATEKEEPER: ...To HERE  -->

peterfpeterson committed:
Remove __future__ imports

These were only used in python 2 and no-op in current python 3

jmborr commented:
@coderabbitai full review

jmborr commented on qt/python/mantidqt/mantidqt/widgets/codeeditor/execution.py:
```diff
@@ -34,39 +32,15 @@
-
-
-def get_future_import_compiler_flags(code_str):
+def get_future_import_compiler_flags(code_str) -> int:
```
are you planning getting rid of this function? It does serve any purpose now, right?

peterfpeterson replied:
It is mostly documentation. I guess I can drop it too

peterfpeterson committed:
Change no-op function to constant/magic number

peterfpeterson committed:
Put back things that import __future__ annotations

This was supposed to be default on in python 3.10, but it isn't
according to https://www.aemonge.com/articles/python/performance/annotations.html

jmborr reviewed (approved):
:ok_hand:

ajjackson commented:
In practice we should be able to remove

```python
from __future__ import annotations
```

when the minimum Python version becomes 3.14.

ajjackson commented:
Seekpath has accepted my pull request; if we like we could ensure that Mantid uses the future-free recipe by pinning `seekpath=2.1.0=pyhd8ed1ab_2`.

I would prefer not to pin this at the Euphonic level; this library is intended to have many downstream users with different pinning situations so we try to be permissive.

ajjackson commented:
Why? Isn't it sufficient for Mantid to pin the appropriate Seekpath?

peterfpeterson commented:
I can play with it some, but I don't know that version/buildnumber specification work that way. I'm suspicious that the short term answer is to select the specific version of seekpath and add a comment that the dependency can go away when euphonic releases a new version using the other build.
